### PR TITLE
[17.0][MIG] subscription_oca: Migration to 17.0

### PR DESCRIPTION
**Pull request by Tisho99**
Standard migration 

Supersedes #1068

T-6329

### CODE REVIEW

**Comment by HaraldPanten** [comment]
@pedrobaeza do you mind assigning this migration to @Tisho99 and close https://github.com/OCA/contract/pull/1068 ? We'll finish this.

**Comment by Tisho99** [comment]
@HaraldPanten @ValentinVinagre 

PR ready for review

**Review by luis-ron — APPROVED** [review]
Functional Review: LGTM 👍🏻

**Review by HaraldPanten — APPROVED** [review]
LGTM

File: subscription_oca/models/sale_subscription_line.py [hunk 1]
```diff
@@ -62,10 +62,10 @@ def _compute_subtotal(self):
             )
 
     price_subtotal = fields.Monetary(
-        string="Subtotal", readonly="True", compute=_compute_subtotal, store=True
+        string="Subtotal", readonly=True, compute="_compute_subtotal", store=True
```
Comment (rousseldenis) [inline subscription_oca/models/sale_subscription_line.py]: @Tisho99 With this change, you can put fields definitions on top.

**Review by rousseldenis — APPROVED** [review]
Minor remark

**Comment by HaraldPanten** [comment]
@rousseldenis Could we merge, then?

THX!

File: subscription_oca/models/sale_order.py [hunk 2]
```diff
@@ -49,7 +49,7 @@ def create_subscription(self, lines, subscription_tmpl):
             rec = self.env["sale.subscription"].create(
                 {
                     "partner_id": self.partner_id.id,
-                    "user_id": self._context["uid"],
+                    "user_id": self._context.get("uid", self.env.uid),
```
Comment (rousseldenis) [inline subscription_oca/models/sale_order.py]: Oops didn't see this:

`self.env.context.get(...)`is better as _context is deprecated since a long time

**Pull request merged**